docs/hub/datasets-streaming.md

```diff
@@ -0,0 +1,158 @@
+
+>![TIP]
+> There is an equivalent filesystem implementation in Rust available in [OpenDAL](https://github.com/apache/opendal)
+
+## Using cURL
+
+Since all files on the Hub are available via HTTP, you can stream files using `cURL`:
+
+```bash
+>>> curl -L https://huggingface.co/datasets/fka/awesome-chatgpt-prompts/resolve/main/prompts.csv | head -n 5
+"act","prompt"
+"An Ethereum Developer","Imagine you are an experienced Ethereum developer tasked with creating...
+"SEO Prompt","Using WebPilot, create an outline for an article that will be 2,000 words on the ...
+"Linux Terminal","I want you to act as a linux terminal. I will type commands and you will repl...
+"English Translator and Improver","I want you to act as an English translator, spelling correct...
+```
+
+Use range requests to access a specific portion of a file:
+
+```bash
+>>> curl -r 40-88 -L https://huggingface.co/datasets/fka/awesome-chatgpt-prompts/resolve/main/prompts.csv
+Imagine you are an experienced Ethereum developer
+```
+
+Stream from private repositories using an [access token](https://huggingface.co/docs/hub/en/security-tokens):
+
+
+```bash
+>>> export HF_TOKEN=hf_xxx
+>>> curl -H "Authorization: Bearer $HF_TOKEN" -L https://huggingface.co/...
+```
+
+## Streaming Parquet
+
+Parquet is a great format for AI datasets. It offers good compression, a columnar structure for efficient processing and projections, and multi-level metadata for fast filtering, and is suitable for datasets of all sizes.
+
+Parquet files are divided in row groups that are often around 100MB each. This lets data loaders and data processing frameworks stream data progressively, iterating on row groups.
+
+### Stream Row Groups
+
+Use PyArrow to stream row groups from Parquet files on Hugging Face:
+
+```python
+import pyarrow.parquet as pq
+
+repo_id = "HuggingFaceFW/finewiki"
+path_in_repo = "data/enwiki/000_00000.parquet"
+
+# Stream the Parquet file row group per row group
+with pq.ParquetFile(f"hf://datasets/{repo_id}/{path_in_repo}") as pf:
+    for row_group_idx in range(pf.num_row_groups):
+        row_group_table = pf.read_row_group(row_group_idx)
+        df = row_group_table.to_pandas()
+```
+
+> ![TIP]
+> PyArrow supports `hf://` paths out-of-the-box and uses `HfFileSystem` automatically
+
+Find more information in the [PyArrow documentation](./datasets-pyarrow).
+
+### Efficient random access
+
+Row groups are further divied into columns, and columns into pages. Pages are often around 1MB and are the smallest unit of data in Parquet, since this is where compression is applied. Accessing pages enables loading specific rows without having to load a full row group, and is possible if the Parquet file has a page index. However not every Parquet frameworks supports reading at the page level. PyArrow doesn't for example, but the `parquet` crate in Rust does:
```

Comment:
add a good visual of Parquet file format, and btw we should have a /docs/hub/parquet page probably...